src/main/java/org/cbioportal/legacy/service/util/SessionServiceRequestHandler.java

```diff
@@ -65,11 +74,17 @@ public String getSessionDataJson(SessionType type, String id) throws Exception {
 
     RestTemplate restTemplate = new RestTemplate();
 
+    String url =
+        UriComponentsBuilder.fromUriString(sessionServiceURL)
+            .pathSegment(type.name())
+            .pathSegment(id)
+            .build()
+            .toUriString();
```

Comment:
nice improvement to use a url builder instead. It avoids potential url formatting issues.